### PR TITLE
chore: set IMAGE to release-manifests

### PR DESCRIPTION
**Pull request by Xunzhuo**
Fixes: #550

image
 

Signed-off-by: bitliu <bitliu@tencent.com>

### CODE REVIEW

File: tools/make/kube.mk [hunk 1]
```diff
@@ -101,4 +103,20 @@ delete-cluster: $(tools/kind) ## Delete kind cluster.
...
+	@curl -sLo release-artifacts/gatewayapi-crds.yaml ${GATEWAY_RELEASE_URL}
+	@echo "Added: release-artifacts/gatewayapi-crds.yaml"
+
+	@mkdir -pv $(OUTPUT_DIR)/manifests/provider
+	@cp -r $(KUBE_PROVIDER_DIR) $(OUTPUT_DIR)/manifests/provider
+	@mkdir -pv $(OUTPUT_DIR)/manifests/infra
+	@cp -r $(KUBE_INFRA_DIR) $(OUTPUT_DIR)/manifests/infra
+	@cd $(OUTPUT_DIR)/manifests/provider/config/envoy-gateway && $(ROOT_DIR)/$(tools/kustomize) edit set image envoyproxy/gateway-dev=$(IMAGE):$(TAG)
+	@$(tools/kustomize) build $(OUTPUT_DIR)/manifests/provider/config/default > release-artifacts/envoy-gateway.yaml
+	@$(tools/kustomize) build $(OUTPUT_DIR)/manifests/infra/config/rbac > release-artifacts/infra-manager-rbac.yaml
```
Comment (arkodg) [inline tools/make/kube.mk:116]: should these intermediate YAML be deleted so they dont get uploaded ?

File: tools/make/kube.mk [hunk 2]
```diff
@@ -101,4 +103,20 @@ delete-cluster: $(tools/kind) ## Delete kind cluster.
...
+	@$(tools/kustomize) build > release-artifacts/install.yaml
+	@echo "Generated: release-artifacts/install.yaml"
+
+	@cp examples/kubernetes/quickstart.yaml release-artifacts/quickstart.yaml
```
Comment (arkodg) [inline tools/make/kube.mk:120]: lets rm `@` from all non echo commands to improve debuggability

File: tools/make/kube.mk [hunk 3]
```diff
@@ -101,4 +103,20 @@ delete-cluster: $(tools/kind) ## Delete kind cluster.
 
 .PHONY: release-manifests
 release-manifests: $(tools/kustomize) ## Generate Kubernetes release manifests.
-	tools/hack/release-manifests.sh $(GATEWAY_API_VERSION) $(TAG)
+	@mkdir -p release-artifacts/
+	@curl -sLo release-artifacts/gatewayapi-crds.yaml ${GATEWAY_RELEASE_URL}
+	@echo "Added: release-artifacts/gatewayapi-crds.yaml"
+
+	@mkdir -pv $(OUTPUT_DIR)/manifests/provider
```
Comment (arkodg) [inline tools/make/kube.mk:110]: can you try and reuse the previous work you did in https://github.com/envoyproxy/gateway/blob/2fb3ca8954705dabbcfa08007b658655fcb60f46/tools/make/kube.mk#L41

File: tools/make/kube.mk (hunk 2 above)
Comment (Xunzhuo) [reply tools/make/kube.mk:120]: Okay

File: tools/make/kube.mk (hunk 1 above)
Comment (Xunzhuo) [reply tools/make/kube.mk:116]: https://github.com/envoyproxy/gateway/blob/main/.github/workflows/release.yaml#L37 we specify the release file, so no need to worry about this.

File: tools/make/kube.mk (hunk 3 above)
Comment (Xunzhuo) [reply tools/make/kube.mk:110]: That will apply the outputs, but right, we can make this more common used.

File: .github/workflows/build_and_test.yaml [hunk 4]
```diff
@@ -64,9 +64,9 @@ jobs:
     - name: Build and Push EG Commit Image
       if: github.event_name == 'push'
       # tag is set to the short SHA of the commit
-      run: make image.push.multiarch PLATFORMS="linux_amd64 linux_arm64"
+      run: make image.push.multiarch PLATFORMS="linux_amd64 linux_arm64" IMAGE=envoyproxy/gateway-dev
```
Comment (arkodg) [inline .github/workflows/build_and_test.yaml:67]: is this change needed or just added it here to make it more explicit ?

Comment (Xunzhuo) [reply .github/workflows/build_and_test.yaml:67]: Not needed, but to make it more explicit:), used for people who want to have a pre release

**Review by arkodg — APPROVED** [review]
🚀
